spec/compiler/semantic/restrictions_spec.cr

```diff
@@ -86,16 +86,27 @@ describe "Restrictions" do
       )) { types["Foo"] }
   end
 
-  it "allows typeof as restriction" do
-    assert_type(%(
-      struct Int32
-        def self.foo(x : typeof(self))
-          x
-        end
+  it "errros if using typeof" do
```

Comment:
"errors"

Reply:
Thanks. Fixed with an amend and force push.